Fix a regression so that a shadow class is not generated if there is a private
dtor.

https://www.riverbankcomputing.com/hg/sip/rev/6b09a6d578e8

diff -r 6209a625ac87 -r 6b09a6d578e8 sipgen/transform.c
--- sipgen/transform.c	Thu Mar 30 09:43:04 2017 +0100
+++ sipgen/transform.c	Sun Apr 02 11:49:58 2017 +0100
@@ -273,7 +273,7 @@
     for (cd = pt->classes; cd != NULL; cd = cd->next)
         if (cd->iff->type == class_iface)
         {
-            if (needsShadow(cd) && !isIncomplete(cd) && canCreate(cd))
+            if (needsShadow(cd) && !isIncomplete(cd) && !isPrivateDtor(cd) && canCreate(cd))
                 setHasShadow(cd);
 
             /* Get the list of visible Python member functions. */



